### DISCUSSION TITLE
Release v0.3: The Basics Update

### DISCUSSION THREAD

**Discussion by minecraftchloe** [Mojang/minecraft-editor#2] [Announcements]
# Release v0.3: The Basics Update
_Our first (humble) release of the Editor in Bedrock Preview - 1.19.80 Previews 🎉_

First things first. You'll notice we refer to this release as v0.3. _Wait! What about v0.1 and v0.2?!_ Never fear, we can explain! Versions 0.1 and 0.2 were internal – v0.1 was prototyping and establishing the foundation, and v0.2 was building a basic set of features and making it ready to share with you. We wanted to make things easier for ourselves, so we're aligning our internal and external version # tracking. 

## What's included

**Availability**
- Keyboard/mouse on Windows PC Bedrock Preview
- Single or multi(creator) (peer-to-peer and dedicated server)
- Editor in a dedicated server only works on Win32 clients at the moment
- Accessible via client-created desktop shortcut

**Launch**
- Create New Project: Terminology change from "worlds" to "projects" 
- Editor creates a new filetype (.mcproject). Exported projects exist in `com.mojang/projectbackups`.
- Import support for: .mcworld, .mcproject, or .mctemplate filetypes ([View documentation for file extensions](https://learn.microsoft.com/minecraft/creator/documents/minecraftfileextensions))

**UI**
- Quick start welcome guide with a concise how-to
- UI settings (`File -> UI settings`) allow you to change the UI Scale, Font, and Theme. These settings persist between projects. On fresh installs, you will need to reconfigure to your preferences. 
- Top menu, action bar, and toolrail have UI elements, keyboard shortcuts, and tooltips
- Status bar: Located on the bottom left and displays position coordinates

**Functionality**
(_The `Known issues` section contains some quirks to be aware of with the functionalities._)

- Swap between Crosshair Mode (precision single-block editing, mob placement, same controls as Creative Mode) and Tool Mode (multiblock editing, with the UI). If there is anything you aren't able to do in Tool Mode (such as placing mobs), swap to Crosshair Mode.
- Create single (freeform, fixed distance, and adjacent) or multiple selections (`Ctrl` between individual selections) using your mouse
- Resize a selection (this will not resize the selection contents, only the selection outline) using the panel (`Size`) or corner gizmos
- Move a selection (this will not move the selection contents, only the selection outline) using the panel (`Origin`) or middle gizmo
- Deselect a selection or paste preview using the panel or `Ctrl + D` 
- Fill a selection using the panel or `Ctrl + F` (shortcut if you already have your block chosen)
- Cut (`Ctrl + X`), copy (`Ctrl + C`), or delete (`Del`) a selection
- Quick paste (`Ctrl + V`) and paste preview (`Ctrl + Shift + V`) the most recent item in your clipboard that you cut or copied
- Grapple: When in Tool Mode, point anywhere with your 3D cursor and hit `G` on your keyboard to quick teleport.
- Undo (`Ctrl + Z`) and redo (`Ctrl + Y`) block transactions in both Tool and Crosshair Modes
- Change dimension 1:1 wherever you are from the `World Options` menu
- Turn off the daylight cycle from the `World Options` menu
- Globally pause or unpause actors from the `World Options` menu
- View your coordinates in the bottom left status bar
- Use slash commands in both Tool and Crosshair Modes ([View documentation for all slash commands](https://learn.microsoft.com/minecraft/creator/documents/commandsintroduction))

**Sharing**
- Export your Editor project as a playable world (.mcworld) to your `com.mojang/projectbackups` directory

**Accessibility**
- Navigate to `File -> Pause screen -> Accessibility.` Making the Editor inclusive and accessible is important to us. We are working closely with the Minecraft accessibility team, and if you notice improvement opportunities we may have missed, please let us know. 

## Known issues
_There are many known issues in our backlog that we are either working on or have plans to address. :) Here is a snippet of some of them! We welcome your opinions and insights to help us shape the solutions -- and also, we invite you to help us find more!_

1. The initial UI settings may be too large depending on if you are using a 1080p or 4k device, or if the orientation of your monitor is landscape or portrait. Adjust to your preference via `File` > `UI Settings`.
2. Available blocks for filling and brush: We're airing on the side of including more than what might optimally work. For example, if you fill a selection with flowers, they will break because of the placement rules. A workaround is to unpause the world to allow them to naturally despawn, else you can /command with `/kill @e [r=#]` with `r` representing the radius and `#` representing the numeric value of the radius.
3. Breaking and despawn: If items break and the world is paused, they will not despawn. Unpause the world to enable ticking, or do a manual `/kill @e [r=#]` to remove. 
4. Gizmos to rotate a selection outline or paste preview don't exist yet. 
5. Undo/redo only applies to blocks, is cleared when you swap dimensions, and does not undo or redo environmental effects or collateral damage from user action (eg. if you fill a field with water and flowers break, undo or redo will not reinstate the flowers). We have quite a few improvements to make with our undo/redo infrastructure and also how blocks that are subject to gravity (eg. sand, gravel) behave! 
6. If you are building for Bedrock: Education, ensure you enable `Education edition` on project creation under `Cheats` in order to render any Education-only blocks and items. 
7. `Create New Project` isn't as intuitive as it could be with the project settings vs. export settings. We have plans to improve it!
8. The gizmo for moving a selection box can get easily hidden depending on what you have selected, resulting in the panel being the only way to move a selection outline.
9. If you are playing peer-to-peer or on a server, `Export as > Playable world` downloads to the host client. This can lead to spamming. Friends don't spam friends, so be kind with your exporting until we rework this a bit.
10. Pausing and unpausing is global and applies to actors only (eg. water will still flow and fire will still spread). If you're editing in multi(creator), confirm with your building partner(s) before unpausing! 
11. Paste Preview is currently just an empty selection outline. We will be iterating on this to contain an actual preview of the contents you want to paste, and also testing out attaching it to the cursor for more precise placement.
12. Clipboard (cut or copy) is a single item and there is currently no visualization of what is in your clipboard.
13. Visualizations: We think it'd be great to be able to see what you're actually going to be painting with a brush. 
14. In order to see new blocks for the 1.20 update, enable `Next Major Update` toggle in `Experiments` when you create your new project.
15. Custom blocks for tools to use: We want to improve the experience for displaying custom blocks in the block selector. 

---

## Helpful feedback areas
View the [Q&A](https://github.com/Mojang/minecraft-editor/discussions/categories/q-a) section of the Discussion for **Editor Team Questions**.

---

## Feedback loops
We welcome any and all feedback! The more you share, the better, and it will directly influence what we build next. We'll do our best to keep up!

---

## Next
There are _a lot_ of functionalities we want to build into the core experience of the Editor. Our short-term priorities, in order, are to maintain stability in Preview, and to release v0.4: The API Update. However, you can expect to see small, incremental iterations (eg. 0.3.X) based on your feedback and the above known issues week to week.

v0.4: The API Update will be focused on exposing the Editor APIs for you to begin building your own Editor extensions. For now we invite you to acquaint yourself with the existing [scripting](https://docs.microsoft.com/en-us/minecraft/creator/scriptapi/) efforts and get familiar with Typescript!

---

## Appreciation
Thanks for being here and trying this out! We can't wait to build this with you!

**Comment by SmokeyStack** [comment]
Amazing work to the team! Can't wait to see the v1.0 release!

**Comment by xKingDark** [comment]
Thanks for making such an amazing feature <3

**Comment by MisledWater79** [comment]
Such a cool addition can't wait to see where it goes!

**Comment by CaptainCooky** [comment]
This is such an exciting Project, and I hope we can all help to make it an amazing feature! Can't wait to see some cool community creations and extensions!

**Comment by AdamRaichu** [comment]
Do the devs have a plan as to when this will be taken out of preview for use with the release version of BE?

**Reply by minecraftchloe** [reply]
Great question - we're going to be in Preview for a TBD amount of time. Eventually, we _will_ graduate out of Preview, though!
